toolchain/check/testdata/interop/cpp/function/void_pointer.carbon

```diff
@@ -0,0 +1,594 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// INCLUDE-FILE: toolchain/testing/testdata/min_prelude/full.carbon
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/interop/cpp/function/void_pointer.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/interop/cpp/function/void_pointer.carbon
+
+// --- non_nullable_param.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp inline '''
+auto foo(void* _Nonnull x) -> void;
+''';
+
+fn F(input: Cpp.void*) {
+  //@dump-sem-ir-begin
+  Cpp.foo(input);
+  //@dump-sem-ir-end
+}
+
+// --- non_nullable_return_value.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp inline '''
+auto foo() -> void* _Nonnull;
+''';
+
+fn F() {
+  //@dump-sem-ir-begin
+  let output: Cpp.void* = Cpp.foo();
+  //@dump-sem-ir-end
+}
+
+// --- nullable_param.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp inline '''
+auto foo(void* x) -> void;
+''';
+
+fn F(input: Cpp.void*) {
+  //@dump-sem-ir-begin
+  Cpp.foo(input);
+  //@dump-sem-ir-end
+}
+
+// --- fail_todo_null_param.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp inline '''
+auto foo(void* x) -> void;
+''';
+
+fn F() {
+  //@dump-sem-ir-begin
+  // CHECK:STDERR: fail_todo_null_param.carbon:[[@LINE+4]]:11: error: call argument of type `Core.Optional(Cpp.void* as Core.OptionalStorage)` is not supported [CppCallArgTypeNotSupported]
+  // CHECK:STDERR:   Cpp.foo(Core.Optional(Cpp.void*).None());
+  // CHECK:STDERR:           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  // CHECK:STDERR:
+  Cpp.foo(Core.Optional(Cpp.void*).None());
+  //@dump-sem-ir-end
+}
+
+// --- nullable_return_value.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp inline '''
+auto foo() -> void*;
+''';
+
+fn F() {
+  //@dump-sem-ir-begin
+  let output: Core.Optional(Cpp.void*) = Cpp.foo();
+  //@dump-sem-ir-end
+}
+
+// --- todo_fail_void.carbon
```

Comment:
In `TypeCompleter::AddNestedIncompleteTypes`, add:

```
    case SemIR::CppVoidType::Kind: {
      // `void` is always incomplete.
      return false;
    }
```

That should get this to fail appropriately.